subsys/logging/log_core.c

```diff
@@ -84,7 +84,9 @@ static inline void msg_finalize(struct log_msg *msg,
 	irq_unlock(key);
 
 	if (panic_mode) {
+		key = irq_lock();
```

Comment:
Have you considered using k_sched_lock() instead?

Reply:
k_shed_lock() won't be enough since logger can be used from interrupts. In terms of SMP i guess it would be better to use spin_lock, right?